File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    let count = 0;
    
    function dfs(result, depth) {
        if (depth >= numbers.length) {
            if (result === target) {
                count += 1;
            }
            return;
        }
        
        dfs(result + numbers[depth], depth + 1);
        dfs(result - numbers[depth], depth + 1);
    }
    
    dfs(0,0);

    return count;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    
    return dfs(0, 0, numbers, target)


def dfs(i, total, numbers, target):

    if i == len(numbers):
        return 1 if total == target else 0
    
    return dfs(i + 1, total + numbers[i], numbers, target) + dfs(i + 1, total - numbers[i], numbers, target) 